video/out/wayland_common.c

```diff
@@ -2022,6 +2106,8 @@ void vo_wayland_uninit(struct vo *vo)
     wl_list_for_each_safe(output, tmp, &wl->output_list, link)
         remove_output(output);
 
+    talloc_free(wl->drm_formats);
```

Comment:
if you parented `drm_formats` to `wl` during (re)allocations then you would not have to manually free it, that's the whole point of talloc